src/BasicWL/ApplicableHeaderTradeSettlement.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tiime\CrossIndustryInvoice\BasicWL;
+
+use Tiime\EN16931\DataType\CurrencyCode;
+use Tiime\EN16931\DataType\Identifier\BankAssignedCreditorIdentifier;
+
+/**
+ * BG-19.
+ */
+class ApplicableHeaderTradeSettlement
+{
+    /**
+     * BT-90.
+     */
+    private ?BankAssignedCreditorIdentifier $creditorReferenceId;
+
+    /**
+     * BT-83.
+     */
+    private ?string $paymentReference;
+
+    /**
+     * BT-6.
+     */
+    private ?CurrencyCode $taxCurrencyCode;
+
+    /**
+     * BT-5.
+     */
+    private CurrencyCode $invoiceCurrencyCode;
+
+    /**
+     * BG-10.
+     */
+    private ?PayeeTradeParty $payeeTradeParty;
+
+    /**
+     * BG-16.
+     */
+    private ?SpecifiedTradeSettlementPaymentMeans $specifiedTradeSettlementPaymentMeans;
+
+    /**
+     * BG-23.
+     *
+     * @var non-empty-array<int, ApplicableTradeTax>
+     */
+    private array $applicableTradeTaxes;
+
+    /**
+     * BG-14.
+     */
+    private ?BillingSpecifiedPeriod $billingSpecifiedPeriod;
+
+    /**
+     * BG-20.
+     *
+     * @var array<int, SpecifiedTradeAllowance>
+     */
+    private array $specifiedTradeAllowance;
+
+    /**
+     * BG-21.
+     *
+     * @var array<int, SpecifiedTradeCharge>
+     */
+    private array $specifiedTradeCharge;
+
+    /**
+     * BT-20-00.
+     */
+    private ?SpecifiedTradePaymentTerms $specifiedTradePaymentTerms;
+
+    /**
+     * BG-22.
+     */
+    private SpecifiedTradeSettlementHeaderMonetarySummation $specifiedTradeSettlementHeaderMonetarySummation;
+
+    /**
+     * BG-3.
+     *
+     * @var array<int, InvoiceReferencedDocument>
+     */
+    private array $invoiceReferencedDocument;
```

Comment:
Should be "private ?InvoiceReferencedDocument $invoiceReferencedDocument;" for cardinality 0..1.